### PR TITLE
 Table Panel: Improvements to text overflow hover

### PR DESCRIPTION
**Pull request by codeincarnate**
Working on improvements for text wrapping hover behavior to elaborate on feedback from #82872

This looks at the length of the displayed value in the table, if it's relatively short (140 characters or less) and contains whitespace this will cause the hovering to wrap. However, if it's not the case the hover will overflow. There's a performance trade-off here in terms of the string length so we may not be getting every possible wrapping scenario perfect, but this seems to work fairly well.

Video in action:

https://github.com/grafana/grafana/assets/199847/3556e3ea-817a-426b-aaba-3e6b9c86c343

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: packages/grafana-ui/src/components/Table/styles.ts [hunk 1]
```diff
@@ -51,7 +52,7 @@ export function useTableStyles(theme: GrafanaTheme2, cellHeightOption: TableCell
         width: overflowOnHover ? 'auto' : undefined,
         height: overflowOnHover ? 'auto' : `${rowHeight - 1}px`,
         minHeight: `${rowHeight - 1}px`,
-        wordBreak: overflowOnHover ? 'break-word' : undefined,
+        wordBreak: overflowOnHover && asCellText ? 'break-word' : undefined,
```
Comment (ahuarte47) [inline packages/grafana-ui/src/components/Table/styles.ts:55]: `isLongString` or `asCellText`? is it a typo?

Comment (codeincarnate) [reply packages/grafana-ui/src/components/Table/styles.ts]: Fixed, thanks! 🙏

File: packages/grafana-ui/src/components/Table/styles.ts [hunk 2]
```diff
@@ -48,10 +49,10 @@ export function useTableStyles(theme: GrafanaTheme2, cellHeightOption: TableCell
 
       '&:hover': {
         overflow: overflowOnHover ? 'visible' : undefined,
-        width: overflowOnHover ? 'auto' : undefined,
-        height: overflowOnHover ? 'auto' : `${rowHeight - 1}px`,
+        width: stringShouldWrap ? ' auto' : 'auto !important',
```
Comment (ahuarte47) [inline packages/grafana-ui/src/components/Table/styles.ts:52]: maybe there is here an extra space: ' auto'

Comment (codeincarnate) [reply packages/grafana-ui/src/components/Table/styles.ts:52]: Fixed!

File: packages/grafana-ui/src/components/Table/DefaultCell.tsx [hunk 3]
```diff
@@ -15,6 +15,8 @@ import { TableStyles } from './styles';
 import { TableCellProps, CustomCellRendererProps, TableCellOptions } from './types';
 import { getCellOptions } from './utils';
 
+const OG_TWEET_LENGTH = 140; // 🙏
```
Comment (baldm0mma) [inline packages/grafana-ui/src/components/Table/DefaultCell.tsx:18]: RIP

File: packages/grafana-ui/src/components/Table/DefaultCell.tsx [hunk 4]
```diff
@@ -61,13 +67,10 @@ export const DefaultCell = (props: TableCellProps) => {
     }
   }
 
+  // console.log({style: cellProps.style, stringShouldWrap, hover});
```
Comment (baldm0mma) [inline packages/grafana-ui/src/components/Table/DefaultCell.tsx:70]: Should this be removed?

**Comment by baldm0mma** [comment]
> [CLA assistant check](https://cla-assistant.io/grafana/grafana?pullRequest=83352) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/grafana/grafana?pullRequest=83352) before we can accept your contribution.**2** out of **3** committers have signed the CLA.✅ codeincarnate✅ baldm0mma❌ Kyle Cunningham
> 
> **Kyle Cunningham** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).
> You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/grafana?pullRequest=83352) it.

@codeincarnate this seems to be blocking drone from running checks... any idea why it thinks "Kyle Cunningham" is someone else?!?!?

**Comment by codeincarnate** [comment]
Hmmm, that's strange 🤔. Wonder if I may have messed up my local git config, I can take a look :eyes

**Comment by codeincarnate** [comment]
Oof, think I may have accidentally broken this. Going to do another PR with the same and close this one

**Pull request closed without merging**